libs/mocks/src/pools.rs

```diff
@@ -64,6 +68,42 @@ pub mod pallet {
 		) {
 			register_call!(move |(a, b, c)| f(a, b, c));
 		}
+
+		pub fn mock_info(
+			f: impl Fn(
+					T::TrancheCurrency,
+				) -> Result<
+					InvestmentInfo<T::AccountId, T::CurrencyId, T::TrancheCurrency>,
+					DispatchError,
+				> + 'static,
+		) {
+			register_call!(f);
+		}
+
+		pub fn mock_balance(f: impl Fn(T::TrancheCurrency, &T::AccountId) -> T::Balance + 'static) {
+			register_call!(move |(a, b)| f(a, b));
+		}
+
+		pub fn mock_transfer(
+			f: impl Fn(T::TrancheCurrency, &T::AccountId, &T::AccountId, T::Balance) -> DispatchResult
+				+ 'static,
+		) {
+			register_call!(move |(a, b, c, d)| f(a, b, c, d));
+		}
+
+		#[allow(non_snake_case)]
+		pub fn mock_InvestmentAccountant_deposit(
+			f: impl Fn(&T::AccountId, T::TrancheCurrency, T::Balance) -> DispatchResult + 'static,
+		) {
+			register_call!(move |(a, b, c)| f(a, b, c));
+		}
+
+		#[allow(non_snake_case)]
+		pub fn mock_InvestmentAccountant_withdraw(
+			f: impl Fn(&T::AccountId, T::TrancheCurrency, T::Balance) -> DispatchResult + 'static,
+		) {
+			register_call!(move |(a, b, c)| f(a, b, c));
+		}
```

Comment:
The application of https://github.com/foss3/runtime-pallet-library/pull/16, @wischli 

Basically the pallet implements `deposit()` and `withdraw()` methods from two different traits. So we need to add `mock_<trait>_method()` to one of these `deposit()/withdraw()`.

In those cases, the `T::AccountId` used as generic for `InvestmentAccountant` was wrongly internally coded.